docs/is_new_stable.py

```diff
@@ -0,0 +1,31 @@
+import sys
+
+import packaging.version
+
+
+def is_new_stable(old_stable, tag):
+    """
+    Return 0 if `tag` should be the new 'stable' docs folder, 1 otherwise.
+    """
+    try:
+        v_t = packaging.version.Version(tag)
+    except packaging.version.InvalidVersion:
+        return 1
+    v_s = packaging.version.Version(old_stable)
+    return 0 if not v_t.is_prerelease and v_t > v_s else 1
+
+
+if __name__ == "__main__":
+    assert is_new_stable("0.2.0", "taggy-mac-tag-tag") != 0
```

Comment:
Will these  `"0.2.0"` need to be modified each time we have a new release?

Reply:
No, the old_stable is passed in as a command line argument, these are just my unit tests. :) I made this script yesterday and now I'm copying it around to a few repos, so it made sense to have the unit tests in the script. And it makes it self evident what the intended behaviour is in different circumstances.

Reply:
Ok, gotcha! Was just trying to figure out all that is happening. Learning a lot about github actions/docs/etc from these recent PRs, even though I don't have enough of a knowledge about it to have useful comments.

Reply:
I'm learning as I go here, but am finally starting to see how the pieces fit together and am also finding it easier to understand why things fail when they fail. I reckon the github actions docs got 50% better since I looked through them earlier in the year. The most pertinent bit is [`GITHUB_REF_NAME`](https://docs.github.com/en/actions/learn-github-actions/environment-variables) which for 'push' events to a branch will be set to the branch name (`main` most of the time for us) and for 'push' events that push a tag it will be set to the tag name. We previously had a sed command that grabbed the ref name from the 'ref', which has a leading 'refs/heads/' for branches and 'refs/tags/' for tags.